### DISCUSSION TITLE
Is it possible to skip the install and synchronization of certain hooks?

### DISCUSSION THREAD

**Discussion by ardelato** [evilmartians/lefthook#715] [General]
**TL;DR** Is it possible to run a `lefthook` hook without the `.git/hooks` directory being modified? 

----

We've recently switched from using `git hooks` combined with `lint-staged` to solely using `lefthook` in our monorepo. This change was implemented to streamline managing and executing multiple linters simultaneously during our CI process. Which by using `lefthook` has made it very easy and clean to do! 🙌 

We then encountered a problem where our CI runners were inadvertently syncing the `git hooks` that affected the auto-commit linter fixes, leading to failures. 

![image](https://github.com/evilmartians/lefthook/assets/22064420/91945ff3-70df-4495-b948-3412dc505ae6)

To address this, we configured `lefthook` to skip all hooks in the CI environment except for the specific ones needed, using a `skip` option in the `run` settings.

```yaml
pre-commit:
    parallel: true
    skip:
        - rebase
        - merge
        - run: test ${CI:-false} = true

post-commit:
    skip:
        - rebase
        - run: test ${CI:-false} = true
    piped: true
    commands: *clear_git_cache

ci:
    parallel: true
    files: git diff origin/master --name-only
```

Despite solving the issue, we are curious if there is a simpler solution. Specifically, we want to know if there's a way selectively execute certain hooks in a CI environment without triggering the entire install script. https://github.com/evilmartians/lefthook/blob/1d100c61a28454d9bcdfa7045958dce1e5d8cab1/internal/lefthook/install.go#L111-L191

**Comment by mrexox** [comment]
I think I can add a flag `--no-auto-install` to the `lefthook run` command if it solves you case. Do you find it useful?

**Reply by ardelato** [reply]
Oh most definitely!

I think adding a flag seems like the most straightforward solution with the least amount of changes needed.

**Reply by azu** [reply]
@mrexox Yes. `--no-auto-install` is useful if exists.

I met similar problem.
In my workflow, I've commited git hooks into `.githooks/` dir for controll commit hooks under the VCS. (I want to manage my git hooks in git, so I commit `pre-commit` files.)
However, `lefthook run` modify `.githooks/pre-commit` files unexpectally.

- Reproduce: https://github.com/azu/make-lefthook

📝 My project install `lefthook` binary into `<project>/bin/` dir in setup steps.
`lefthook`'s built-in [`pre-commit`](https://github.com/evilmartians/lefthook/blob/5c6d8ae2a6d9884ee8823539d3600e97da89a6f1/internal/templates/hook.tmpl#L16) does not resolve the `<project>/bin/` dir. Instead of it, My [.githooks/pre-commit](https://github.com/azu/make-lefthook/blob/382d8993d9add85cbef2aff53fd293e996cad042/.githooks/pre-commit#L6) resolve the `<project>/bin/` path.

**Reply by mrexox** [reply]
`lefthook run pre-commit --no-auto-install` should be available in 1.6.11 release

**Reply by azu** [reply]
Thanks!

I've updated example repo and it seem work fine!
https://github.com/azu/make-lefthook

**Comment by ardelato** [comment]
@mrexox The `--no-auto-install` flag worked perfectly! Thank you!

That said, I am okay with closing this discussion if you are as well.

**Comment by cgaube** [comment]
Could this be added to the main .lefthook config file ? 

everytime i execute `github commit` it triggers a `lefthook run` and tries to re-install the hooks which to me is not really needed aswe know it is already setup  as it is being triggered by the commit hook :) 

Created feature-request
https://github.com/evilmartians/lefthook/issues/840
